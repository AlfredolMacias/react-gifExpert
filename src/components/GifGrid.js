import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
// import { getGif } from '../helpers/getGifs';
import PropTypes from 'prop-types'

export const GifGrid = ( { category } ) => {

    // const [images, setImages] = useState([]);
    const {data:images,  loading } = useFetchGifs( category );

    return (
        <>
        <h3 className="card-grid animate__animated animate__fadeIn">{ category }</h3>
        <div className="card-grid animate__animated animate__fadeIn">
           { loading && <p className="animate__animated animate__flash">Loading</p>}
            {
                images.map( img  => (
                    <GifGridItem 
                    key = { img.id }
                    { ...img }/>
                    ))
                }
           
        </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}
