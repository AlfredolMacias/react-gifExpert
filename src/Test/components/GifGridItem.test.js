import React from 'react';
import { shallow } from 'enzyme'
// import { GifGridItem } from '../../components/GifGridItem';
import { GifGridItem } from '../../components/GifGridItem'
// import { GifGridItem } from '../../components/GifGridItem';

describe('Primera prueba sobre GIfGridItem', () => {
    
    const title = "Titulo";
    const url   = "https://localhost/urlPrueba";
    
    const wrapper = shallow(<GifGridItem title={ title } url={ url } />);

    test('Debe mostrar el contenido de GifGridItem correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    })
    
    test('Debe tener un parrafo con el title ', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
    })

    test('Debe tener una imagen con la url y alt correcta', () => {
        const img = wrapper.find('img');
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    })

    test('El Div debe contener la clase animated_fadeIn', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect( className.includes(' animate__fadeIn') ).toBe( true );
    })
    
    
    
})
