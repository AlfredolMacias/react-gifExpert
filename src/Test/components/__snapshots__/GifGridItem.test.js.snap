// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primera prueba sobre GIfGridItem Debe mostrar el contenido de GifGridItem correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo"
    src="https://localhost/urlPrueba"
  />
  <p>
    Titulo
  </p>
</div>
`;
